### PR TITLE
Handle batched TaskExecutionEvent reasons

### PR DESCRIPTION
**Pull request by andrewwdye**
# TL;DR
Handle newly added `TaskExecutionEvent.reasons` field as part of https://github.com/flyteorg/flyteidl/pull/443.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This change updates admin to process the recently added `TaskExecutionEvent.reasons` field, included to support sending batched reason events. This reduces load on both propeller and admin. Admin favors the `reasons` field but falls back to the `reason` field and prior behavior if empty.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3825

## Follow-up issue
N/A

### CODE REVIEW

File: pkg/repositories/transformers/task_execution_test.go [hunk 1]
```diff
@@ -569,6 +656,206 @@ func TestUpdateTaskExecutionModelRunningToFailed(t *testing.T) {
...
+		StartedAt:              &taskEventOccurredAt,
+		TaskExecutionCreatedAt: &taskEventOccurredAt,
+		TaskExecutionUpdatedAt: &taskEventOccurredAt,
+	}
+
+	occuredAt := taskEventOccurredAt.Add(time.Minute)
+	occuredAtProto, err := ptypes.TimestampProto(occuredAt)
+	assert.Nil(t, err)
+	secondOccuredAt := taskEventOccurredAt.Add(time.Minute * 2)
+	secondOccuredAtProto, err := ptypes.TimestampProto(secondOccuredAt)
```
Comment (katrogan) [inline pkg/repositories/transformers/task_execution_test.go:799]: check err here?

**Review by katrogan — DISMISSED** [review]

File: pkg/repositories/transformers/task_execution_test.go (hunk 1 above)
Comment (andrewwdye) [reply pkg/repositories/transformers/task_execution_test.go:799]: Turns out the linter caught this too :)

**Review by katrogan — APPROVED** [review]
